### DISCUSSION TITLE
Best practices for platform-specific LFS files?

### DISCUSSION THREAD

**Discussion by ahernsean** [git-lfs/git-lfs#5395] [Q&A]
My team would like to adopt Git LFS for the storage of large binary files, but many of them are platform specific. That is, if the build is happening on Linux, we need one set of files. If it's on Windows, it's a different set of files. If this were a small set, I wouldn't worry too much about it. But on each platform, it's about 17 GB. The combined 34 GB would be a huge payload to consume every time, especially if only half of it is needed for any particular build.

Any suggestions about how to organize this? My original thought was to create two Git LFS repos, one for Linux and one for Windows (though there are some shared files I've considered putting in a third LFS repo) and then have the build setup clone/pull the right one. But there's a good amount of complexity in making that happen. I have also considered sparse checkouts, but that would require dev training (and I heard a rumor that LFS doesn't necessarily work with sparse checkouts).

What have your experiences been with managing LFS files that need to change per-platform (or per- anything else)? What are the best practices? What do you recommend?

**Comment by bk2204** [comment]
One possibility is to use submodules.  You can initialize one for the shared files and one for each of the OSes.  You may want to tag them identically from time to time to ensure that it's easy to have them stay in sync.

You could also set `lfs.fetchexclude` to `*` and then run `git lfs pull -I linux` (or `windows`) as appropriate.  That's relatively easy to do, and you can set that value in `.lfsconfig` in the root of your repository to make it the default.

It's true that at the moment, Git LFS doesn't work with sparse checkout.  That might be a good idea if we did support it, but we don't right now.
